wagtail/tests/test_management_commands.py

```diff
@@ -622,101 +633,129 @@ def setUp(self):
+        # should stay as they are attached to an in progress workflow
+        self.assertRevisionExists(revision)
 
-        approved_revision = self.page.save_revision(
+        # If workflow is disabled at some point after that, the revision should
+        # be deleted
+        with override_settings(WAGTAIL_WORKFLOW_ENABLED=False):
+            self.run_command()
+            self.assertRevisionNotExists(revision)
+
+    def test_revisions_with_approve_go_live_not_purged(self):
+        revision = self.object.save_revision(
             approved_go_live_at=timezone.now() + timedelta(days=1)
         )
 
-        self.page.save_revision()
+        # Save a new revision to ensure that the approved revision
+        # is not the latest one
+        self.object.save_revision()
 
         self.run_command()
 
-        self.assertIn(
-            approved_revision, Revision.page_revisions.filter(object_id=self.page.id)
-        )
+        self.assertRevisionExists(revision)
 
     def test_purge_revisions_with_date_cutoff(self):
+        old_revision = self.object.save_revision()
 
-        old_revision = self.page.save_revision()
-
-        self.page.save_revision()
+        self.object.save_revision()
 
         self.run_command(days=30)
 
         # revision should not be deleted, as it is younger than 30 days
-        self.assertIn(
-            old_revision, Revision.page_revisions.filter(object_id=self.page.id)
-        )
+        self.assertRevisionExists(old_revision)
 
         old_revision.created_at = timezone.now() - timedelta(days=31)
         old_revision.save()
 
         self.run_command(days=30)
 
         # revision is now older than 30 days, so should be deleted
-        self.assertNotIn(
-            old_revision, Revision.page_revisions.filter(object_id=self.page.id)
-        )
+        self.assertRevisionNotExists(old_revision)
+
+
+class TestPurgeRevisionsCommandForSnippets(TestPurgeRevisionsCommandForPages):
+    def get_object(self):
+        return FullFeaturedSnippet.objects.create(text="Hello world!")
+
+
+class TestPurgeRevisionsCommandForPagesWithPagesOnly(TestPurgeRevisionsCommandForPages):
+    base_options = {"pages": True}
+
+
+class TestPurgeRevisionsCommandForPagesWithNonPagesOnly(
+    TestPurgeRevisionsCommandForPages
+):
+    base_options = {"non_pages": True}
+
+    def assertRevisionNotExists(self, revision):
+        # Page revisions won't be purged if only non_pages is specified
+        return self.assertRevisionExists(revision)
+
+
+class TestPurgeRevisionsCommandForSnippetsWithNonPagesOnly(
+    TestPurgeRevisionsCommandForSnippets
+):
+    base_options = {"non_pages": True}
+
+
+class TestPurgeRevisionsCommandForSnippetsWithPagesOnly(
+    TestPurgeRevisionsCommandForSnippets
+):
+    base_options = {"pages": True}
+
+    def assertRevisionNotExists(self, revision):
+        # Snippet revisions won't be purged if only pages is specified
+        return self.assertRevisionExists(revision)
```

Comment:
Was thinking about adding a couple other cases that test with both `{"pages": True, "non_pages": True}`, but the implementation code itself uses `if pages == non_pages`, so it's exactly the same whether both are `True` or both are `False`.

I don't think it's worth it, but I can add it if you want.

Reply:
I'd say it is not worth it.